### ISSUE TITLE
Nonce attribute does not get set on style tags of CSS Modules and imported styles 

### ISSUE THREAD

**Issue by abhic91** [vercel/next.js#35389]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
 Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.2.0: Sun Nov 28 20:29:10 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T8101
    Binaries:
      Node: 14.17.2
      npm: 8.1.2
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 12.1.0
      react: 17.0.2
      react-dom: 17.0.2
```

### How are you deploying your application? (if relevant)

Netlify

### Describe the Bug

I am using Material UI and CSS modules. The global styles imported in _app.tsx and css modules and also styles for elements that are later animated in fail to load as they are blocked by CSP because the `nonce` attribute is not getting those styles. 

### Expected Behavior

The styles generated by css modules, styles for elements that are not visible on load and imported styles should have `nonce` set on them 

### To Reproduce

This is `_document.tsx`

```
MyDocument.getInitialProps = async (ctx) => {
    const originalRenderPage = ctx.renderPage;
      const cache = createEmotionCache();
      const { extractCriticalToChunks } = createEmotionServer(cache);
      const nonce = nanoid();
      let contentSecurityPolicy = '';
      if (process.env.NODE_ENV === 'production') {
        contentSecurityPolicy = `default-src 'self'; style-src-elem 'self' 'nonce-${nonce}'; style-src 'self' 'nonce-${nonce}'; img-src 'self' data:;`;
      } else {
        contentSecurityPolicy = `default-src 'self'; style-src-elem 'unsafe-inline'; script-src 'self' 'unsafe-eval' 'unsafe-inline';`;
      }
      ctx.res?.setHeader('Content-Security-Policy', contentSecurityPolicy);
    
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: (App: any) =>
            function EnhanceApp(props) {
              return <App emotionCache={cache} {...props} />;
            },
        });
    
      const initialProps = await Document.getInitialProps(ctx);
      const emotionStyles = extractCriticalToChunks(initialProps.html);
      const emotionStyleTags = emotionStyles.styles.map((style) => (
        <style
          nonce={nonce}
          data-emotion={`${style.key} ${style.ids.join(' ')}`}
          key={style.key}
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: style.css }}
        />
      ));
    
      return {
        ...initialProps,
        emotionStyleTags,
      };
```
The component
```
export default class MyDocument extends Document {
  render() {
    const nonce = (this.props as any).nonce;
    console.log((this.props as any).emotionStyleTags);
    return (
      <Html lang="en">
        <Head nonce={nonce}>
          {/* PWA primary color */}
          <meta name="theme-color" content={theme.palette.primary.main} />
          {/* Inject MUI styles first to match with the prepend: true configuration. */}
          {(this.props as any).emotionStyleTags}
        </Head>
        <body>
          <Main />
          <NextScript nonce={nonce} />
        </body>
      </Html>
    );
  }
}
```

**Comment by balazsorban44** [comment]
This issue has been automatically closed because it wasn't verified against next@canary. If you think it was closed by accident, please leave a comment. If you are running into a similar issue, please open a new issue with a reproduction. Thank you.
